src/main/java/org/opentripplanner/graph_builder/module/osm/OSMDatabase.java

```diff
@@ -645,7 +650,8 @@ private void processMultipolygonRelations() {
                     // relation includes way which does not exist in the data. Skip.
                     continue RELATION;
                 }
-                for (Long nodeId : way.getNodeRefs()) {
+                for (TLongIterator iter = way.getNodeRefs().iterator(); iter.hasNext(); ) {
+                    long nodeId = iter.next();
```

Comment:
I think a while loop reads better in this case - but no big deal.

Reply:
True, the while loop itself reads better, but a while loop leaves the iterator in the containing scope. I like that the iterator goes out of scope when the for loop completes. The ideal would be `for (int nodeId : way.getNodeRefs().asArray())` but unfortunately Trove prevents you from seeing the internal array in a `TIntArrayList`.

Reply:
I guess this is the idiomatic Trove way:
```
                boolean allNodesPresent = way.getNodeRefs().forEach(nodeId -> {
                    if (nodesById.containsKey(nodeId)) {
                        return false;
                    } else {
                        MapUtils.addToMapSet(areasForNode, nodeId, way);
                        return true;
                    }
                });
                if (!allNodesPresent) {
                    // this area is missing some nodes, perhaps because it is on
                    // the edge of the region, so we will simply not route on it.
                    continue RELATION;
                }
```
Which is even more verbose. I just changed it to a while loop, the scope only continues about 5 lines past the end of the loop.